buildSrc/src/main/groovy/org/elasticsearch/gradle/test/NodeInfo.groovy

```diff
@@ -177,6 +177,9 @@ class NodeInfo {
             javaVersion = 8
         } else if (nodeVersion.onOrAfter("6.2.0") && nodeVersion.before("6.3.0")) {
             javaVersion = 9
+        } else if (project.inFipsJvm && nodeVersion.onOrAfter("6.3.0") && nodeVersion.before("6.4.0")) {
```

Comment:
I'm sure I'm missing something, but wouldn't this only change the version that we try to start on, but not stop the node from attempting to start?

Reply:
Correct. The intention is to not mute the tests but keep running them even when we're in a FIPS JVM in CI. The way to achieve this is to make sure that older ES version ( not supporting fips ) nodes start with a non fips java version

Reply:
This is not at all clear from the code. Can you please add a comment explaining it? If I understand correctly, by adding this other elseif condition, non fips testing will fall through and continue using the RUNTIME_JAVA_HOME? But isn't that a fips jvm in this case?

Reply:
> This is not at all clear from the code. Can you please add a comment explaining it?

Sure thing. I tried to capture it in `// Versions before 6.4.0 cannot be run in a FIPS 140 JVM` but I agree that's not very clear. 

> If I understand correctly, by adding this other elseif condition, non fips testing will fall through and continue using the RUNTIME_JAVA_HOME? But isn't that a fips jvm in this case?

Not sure I follow your thought. When RUNTIME_JAVA_HOME is a fips JVM, `project.inFipsJvm` will also be true. 
In summary: 

- When running in a non FIPS JVM
  - Nothing changes from the previous behavior
  - We run < 6.2.0 nodes with Java 8
  - We run > 6.2.0 and < 6.3.0 nodes with Java 9
  - We run > 6.3.0 nodes with RUNTIME_JAVA_HOME ( non FIPS )

- When running in a FIPS JVM
  - `project.inFipsJvm` is true
  - We run < 6.2.0 nodes with Java 8 (non fips)
  - We run > 6.2.0 and < 6.3.0 nodes with Java 9 (non FIPS)
  - We run > 6.3.0 and < 6.4.0 nodes with Java 10 (non FIPS)
  - We run > 6.4.0 nodes with RUNTIME_JAVA_HOME ( which is FIPS but > 6.4 nodes can run fine in a FIPS JVM)

Does this make more sense ? Bear with me if I've missed your point entirely.

Reply:
Thanks, that is more clear.

Reply:
@rjernst I updated the comment, let me know if this is clear enough, thanks !